ci-operator/config/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master.yaml

```diff
@@ -0,0 +1,52 @@
+base_images:
+  base:
+    cluster: https://api.ci.openshift.org
+    name: origin-v4.0
+    namespace: openshift
+    tag: base
+binary_build_commands: make
+canonical_go_repository: github.com/openshift/cloud-credential-operator
+images:
+- dockerfile_path: Dockerfile
+  from: base
+  to: cloud-credential-operator
+resources:
+  '*':
+    limits:
+      cpu: '2'
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  cluster: https://api.ci.openshift.org
+  name: origin-v4.0
+  namespace: openshift
+  tag: ''
+  tag_overrides: {}
+build_root:
+  #  image_stream_tag:
```

Comment:
Why are you building your own base image?

Reply:
We have a few build time dependencies (gomock, mockgen) and via the docs it looked like this was the correct way to handle that (https://github.com/openshift/ci-operator/blob/master/CONFIGURATION.md#build_root), it's what we did in Hive as well. Our image is based on openshift/origin-release:golang-1.10.

Is this incorrect use of the config and is there a better way?

Reply:
I think I found a solution, the deps in question are for generation which shouldn't be done for a CI build, it should just use the committed code. Added some make targets for CI that skip generation, and I'm now able to use the standard base image.